library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -122,6 +160,38 @@ internal class TransactionPayloadFragment :
         activity?.invalidateOptionsMenu()
     }
 
+    @RequiresApi(Build.VERSION_CODES.KITKAT)
+    @SuppressLint("DefaultLocale")
+    private fun viewBodyExternally() {
+        viewModel.transaction.value?.let {
+            val intent = Intent(Intent.ACTION_CREATE_DOCUMENT).apply {
+                    addCategory(Intent.CATEGORY_OPENABLE)
+                    putExtra(Intent.EXTRA_TITLE, "$DEFAULT_FILE_PREFIX${System.currentTimeMillis()}")
+                    type = "*/*"
+                }
+                if (intent.resolveActivity(requireActivity().packageManager) != null) {
+                    startActivityForResult(intent, GET_FILE_FOR_SAVING_REQUEST_CODE)
+                } else {
+                    Toast.makeText(
+                        requireContext(), R.string.chucker_save_failed_to_open_document,
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, resultData: Intent?) {
+        if (requestCode == GET_FILE_FOR_SAVING_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            val uri = resultData?.data
+            if (uri != null) {
+                val transaction = viewModel.transaction.value
+                fileSaverTask = FileSaverTask(this).apply {
+                    execute(Triple(type, uri, transaction!!))
```

Comment:
I don't like forcing non-null, since we really can't be 100% sure about value of the `LiveData` being not null. 
Let's move `val transaction` outside `if` and change the check to 
`if (uri != null && transaction !=null)`